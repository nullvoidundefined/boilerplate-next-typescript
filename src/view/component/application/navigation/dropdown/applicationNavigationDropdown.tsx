import styled from "styled-components";
import {
  APPLICATION_COLOR,
  APPLICATION_DIMENSION,
} from "../../../../../constant";

interface ApplicationNavigationDropdownProps {
  isOpen: boolean;
  heightOffset: number;
  onLinkClick: () => void;
}

interface ApplicationNavigationDropdownContentProps {
  heightOffset: number;
}

const ApplicationNavigationDropdownContainer = styled.div`
  position: absolute;
  top: 0;
  width: 100%;
  max-width: ${APPLICATION_DIMENSION.MINIMUM_WIDTH}px;
`;

const ApplicationNavigationDropdownContent = styled.div<ApplicationNavigationDropdownContentProps>`
  background-color: ${APPLICATION_COLOR.NAVIGATION_DROPDOWN_BACKGROUND};
  height: ${({ heightOffset }) => {
    return `calc(100vh - ${heightOffset}px)`;
  }};
`;

const ApplicationNavigationDropdown = ({
  isOpen,
  heightOffset,
}: ApplicationNavigationDropdownProps) => {
  return isOpen ? (
    <ApplicationNavigationDropdownContainer>
      <ApplicationNavigationDropdownContent heightOffset={heightOffset}>
        Navigation Dropdown
      </ApplicationNavigationDropdownContent>
    </ApplicationNavigationDropdownContainer>
  ) : null;
};

export { ApplicationNavigationDropdown };
