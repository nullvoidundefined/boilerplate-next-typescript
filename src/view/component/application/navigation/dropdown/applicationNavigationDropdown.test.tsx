import "jest-styled-components";

import { ApplicationNavigationDropdown } from "./applicationNavigationDropdown";
import renderer from "react-test-renderer";
import { noop } from "lodash";

it("renders correctly", () => {
    const props = {
        isOpen: true,
        heightOffset: 0,
        onLinkClick: () => noop(),
    };
    const component = renderer
        .create(<ApplicationNavigationDropdown {...props} />)
        .toJSON();
    expect(component).toMatchSnapshot();
});
