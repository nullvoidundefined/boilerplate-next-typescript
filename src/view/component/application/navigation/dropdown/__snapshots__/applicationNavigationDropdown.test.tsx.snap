// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: absolute;
  top: 0;
  width: 100%;
  max-width: 390px;
}

.c1 {
  background-color: green;
  height: calc(100vh - 0px);
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    Navigation Dropdown
  </div>
</div>
`;
